src/sympc/session/session.py

```diff
@@ -204,12 +205,64 @@ def przs_generate_random_share(
             shape=shape,
         )
 
-        # It has encoder_precision = 0 such that the value would not be encoded
-        share = ShareTensor(
-            data=current_share - next_share,
-            session_uuid=self.uuid,
-            config=Config(encoder_precision=0),
+        if self.protocol.share_class == ShareTensor:
+            # It has encoder_precision = 0 such that the value would not be encoded
+            share = ShareTensor(
+                data=current_share - next_share,
+                session_uuid=self.uuid,
+                config=Config(encoder_precision=0),
+            )
+        else:
```

Comment:
add an el if condition here and may be, raise an error if share_class is not set.

Reply:
Yeah ,that is in the protocol metaclass , which does not allow for protocol class creation without share_class attribute.https://github.com/OpenMined/SyMPC/blob/16fa17ab61fc37f0aba2f565bda812ae55eb1cdc/src/sympc/protocol/protocol.py#L36